### ISSUE TITLE
How do I modify so it accepts username instead of email?

### ISSUE THREAD

**Issue by dannychase2004** [PHPAuth/PHPAuth#211]
I need to disable the email filter validation so that a person can just sign on with a username, but when I open the file it is read-only. What's the recommended way of editing this?

**Comment by cuonic** [comment]
Making PHPAuth compatible with usernames would require rewriting many parts of the current library and database layout. But whatever happens you you still be able to edit the class however you like, check your file permissions.

**Comment by dannychase2004** [comment]
Composer is what was really messing me up, so I just created new .php files for Auth and Config, and copied/pasted the contents into them.. Then went through every function to make username the key instead of email address, but I kept email address as a secondary field. You're right, it was significant. I'm wondering, if it wasn't for the fact that it's validated, if removing the validation would have been the only thing I really needed to do? If so, it might make sense to make it part of the configuration so that if someone wants to use username instead of emailaddr, they can just turn off validation and it should work.
